Insights/Insights.swift

```diff
@@ -10,174 +10,88 @@ import Foundation
-  ///
-  /// TODO: For this version, the view event is not supported
-  private func view(params: [String: Any]) {
-    process(event: Event(params: params, event: API.Event.view))
-  }
-  
-  private func process(event: Event) {
-    events.append(event)
-    sync(event: event)
-    serialize()
-  }
-  
-  @objc func flushEvents() {
-      flush(self.events)
-  }
-  
-  private func flush(_ events: [Event]) {
-    logger.debug(message: "Flushing remaining events")
-    events.forEach(sync)
-  }
-
-  private func sync(event: Event) {
-    logger.debug(message: "Syncing \(event)")
-    webservice.sync(event: event) {[weak self] err in
-
-      // If there is no error or the error is from the Analytics we should remove it. In case of a WebserviceError the event was wronlgy constructed
-      let webServiceError = err as? WebserviceError
-      if err == nil || webServiceError != nil {
-        self?.remove(event: event)
-      }
+    
+    /// Access an already registered `Insights` without having to pass the `apiKey` and `appId`. If the application was not register before, it will return a nil value.
+    /// - warning: If more than one application has been registered, an undetermined instance of `Insights` will be returned.
+    ///
+    
+    public static var shared: Insights? {
+        return insightsMap.first?.value
     }
-  }
-
-  private func remove(event: Event) {
-    events = events.filter { $0 != event }
-    serialize()
-  }
-
-  func serialize() {
-    if let file = LocalStorage<[Event]>.filePath(for: credentials.indexName) {
-      LocalStorage<[Event]>.serialize(events, file: file)
-    } else {
-      logger.debug(message: "Error creating a file for \(credentials.indexName)")
+    
+    /// Access an already registered `Insights` via its `appId`. If the application was not register before, it will return a nil value.
+    /// - parameter  appId: The appId of application that is being tracked
+    ///
+    public static func shared(appId: String) -> Insights? {
+        return insightsMap[appId]
     }
-  }
-
-  private func deserialize() {
-    guard let filePath = LocalStorage<[Event]>.filePath(for: credentials.indexName) else {
-      logger.debug(message: "Error reading a file for \(credentials.indexName)")
-      return
+    
+    public var loggingEnabled: Bool = false {
+        didSet {
+            eventsController.logger.enabled = loggingEnabled
+        }
     }
-    self.events = LocalStorage<[Event]>.deserialize(filePath) ?? []
-  }
-
-  deinit {
-    flushTimer.invalidate()
-  }
-}
-
-@objcMembers public class Credentials: NSObject {
-
-  let appId: String
-  let apiKey: String
-  let indexName: String
-
-  init(appId: String, apiKey: String, indexName: String) {
-    self.appId = appId
-    self.apiKey = apiKey
-    self.indexName = indexName
-    super.init()
-  }
-}
-
-public enum InsightsException: Error {
-  case credentialsNotFound(String)
+    
+    private let eventsController: EventsController
```

Comment:
The type can be the abstract `EventsProcessor`. That can be helpful for DI in case we want to mock that later on